### DISCUSSION TITLE
Get Client in Logout

### DISCUSSION THREAD

**Discussion by honzapatCZ** [ory/hydra#2468] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
Well, I wanted to make the logout endpoint consider some stuff in the client metadata(eg firstparty) to skip the UI even when its Client initiated request.

**Describe the solution you'd like**
I suppose extending the endpoint for passing along the client too(which is there already so idk why its not passed)

**Describe alternatives you've considered**
If I would knew where I could fetch session_id data I would prob. use that

**Comment by vinckr** [comment]
You mean this step of the [logout flow](https://www.ory.sh/hydra/docs/concepts/logout)?

> 3. The logout provider uses the challenge query parameter to fetch metadata about the request. The logout provider may choose to show a UI where the user has to accept the logout request. Alternatively, the logout provider MAY choose to silently accept the logout request.

> The server response is a JSON object with the following keys:
```
{
  // The user for whom the logout was request.
  "subject": "user-id",

  // The login session ID that was requested to log out.
  "sid": "abc..",

  // The original request URL.
  "request_url": "https://hydra/oauth2/sessions/logout?id_token_hint=...",

  // True if the request did not include a id_token_hint. False otherwise.
  "rp_initiated": true | false
}
```

Can you give me some details what you mean by passing the client along? 
The sid is part of the response, would that work?

**Reply by honzapatCZ** [reply]
Yep, exactly, I know about the sid, but how can I get client info fram that? 
Also looking at source the client is already somehow fetched, so all Iam asking for is having it in the response in case it's rp_initiated and that would also possibly make rp_initiated redundant since in case it is rp_initiated the client would be set if not it would be null

**Comment by pommelinho** [comment]
I would suggest to use https://www.ory.sh/hydra/docs/reference/api/#lists-all-consent-sessions-of-a-subject to get all sessions of the subject. Then you could just filter the result for the login_session_id corresponding to the sid. Then you would get the corresponding client information.

However I also would like it to get the client info right away from hydra when making the logout request

**Reply by honzapatCZ** [reply]
This seems like a really bad idea and also very performance consuming

**Reply by pommelinho** [reply]
As I said I am also interested in having the client info already in the response of the logout request.
However this is the only workaround i currently see to get this information :(

**Reply by honzapatCZ** [reply]
@aeneasr  worth a PR?

**Comment by aeneasr** [comment, marked as answer]
@honzapatCZ @pommelinho we can not include the ClientID because a single login session (sid) can be linked to multiple clients because login and consent are two separate things.

We currently do not keep track of what clients use a sid so this would need some work.

**Reply by honzapatCZ** [reply]
Ye, but if its RP initiated, we know which RP/Client initiated it right?

**Reply by aeneasr** [reply]
Right, in that case you have access to `id_token_hint` which contains the client_id as the audience. We could probably decode the claims and add them to the payload to make this process easier!

**Reply by honzapatCZ** [reply]
where exactly is the id_token_hint

**Reply by aeneasr** [reply]
https://github.com/ory/hydra/discussions/2468#discussioncomment-614230:

`"request_url": "https://hydra/oauth2/sessions/logout?id_token_hint=...",`

**Reply by pommelinho** [reply]
@honzapatCZ  ok when having the client_id. this would mean that you then would use https://www.ory.sh/hydra/docs/reference/api/#get-an-oauth-20-client to get the client and its metadata, right?

**Reply by honzapatCZ** [reply]
> [#2468 (comment)](https://github.com/ory/hydra/discussions/2468#discussioncomment-614230):
> 
> `"request_url": "https://hydra/oauth2/sessions/logout?id_token_hint=...",`

OK, but that is an ID token hint, so I would have to parse that and get the client from there.

If I make a PR and make it return a client also if its rp_initiated?

**Reply by aeneasr** [reply]
Makes sense!

**Reply by honzapatCZ** [reply]
> Makes sense!

Wait there already is client in the definition
https://github.com/ory/hydra/blob/24f91ab7038e4c612db5af86ff0f3ee5eb03c878/consent/types.go#L421

**Comment by vinckr** [comment]
Hey @honzapatCZ, just checking in if your problem still persists. 
I will mark this as answered for now, but feel free to come back to the discussion any time.
Thanks a ton :bee:
